build.gradle

```diff
@@ -1082,7 +1082,7 @@ subprojects {
         }
 
         // Create a task for each JUnit test class whose name is the same as the JUnit class name.
-        sourceSets.test.allJava.filter { it.path.contains('test/junit') }.forEach { file ->
+        sourceSets.test.allJava.filter { it.path.contains('test/junit') || it.path.contains('test\\junit')  }.forEach { file ->
```

Comment:
Can you combine this into one, maybe `'test[/\\]junit'?

Reply:
Hello Professor, `contains` in Groovy only checks simple String. It has to use `matches` to check a regex.

Reply:
Let's use a regex, then, that's probably shorter and cleaner than this.

Reply:
I see that originally you had '.*test[\\/]+junit.*'. I think the `+` is not needed. Maybe add a comment to make the pattern easier to read.

Reply:
I updated the regex to `.*test[\\/]junit.*` because a single backslash is an escape character in a path, and it needs `\\` as a backslash